In [2]:
import pandas as pd


In [8]:
#Load data
df = pd.read_csv('diabetes_dataset.csv')

#Read first few lines
print(df.head())

   year  gender   age location  race:AfricanAmerican  race:Asian  \
0  2020  Female  32.0  Alabama                     0           0   
1  2015  Female  29.0  Alabama                     0           1   
2  2015    Male  18.0  Alabama                     0           0   
3  2015    Male  41.0  Alabama                     0           0   
4  2016  Female  52.0  Alabama                     1           0   

   race:Caucasian  race:Hispanic  race:Other  hypertension  heart_disease  \
0               0              0           1             0              0   
1               0              0           0             0              0   
2               0              0           1             0              0   
3               1              0           0             0              0   
4               0              0           0             0              0   

  smoking_history    bmi  hbA1c_level  blood_glucose_level  diabetes  
0           never  27.32          5.0                  10

In [12]:
#Summary statistics
df.describe(include='all')

year  gender            age location  race:AfricanAmerican  \
count   100000.000000  100000  100000.000000   100000         100000.000000   
unique            NaN       3            NaN       55                   NaN   
top               NaN  Female            NaN     Iowa                   NaN   
freq              NaN   58552            NaN     2038                   NaN   
mean      2018.360820     NaN      41.885856      NaN              0.202230   
std          1.345239     NaN      22.516840      NaN              0.401665   
min       2015.000000     NaN       0.080000      NaN              0.000000   
25%       2019.000000     NaN      24.000000      NaN              0.000000   
50%       2019.000000     NaN      43.000000      NaN              0.000000   
75%       2019.000000     NaN      60.000000      NaN              0.000000   
max       2022.000000     NaN      80.000000      NaN              1.000000   

           race:Asian  race:Caucasian  race:Hispanic     race:Other  \
count   100000.000000   100000.000000   100000.00000  100000.000000   
unique            NaN             NaN            NaN            NaN   
top               NaN             NaN            NaN            NaN   
freq              NaN             NaN            NaN            NaN   
mean         0.200150        0.198760        0.19888       0.199980   
std          0.400114        0.399069        0.39916       0.399987   
min          0.000000        0.000000        0.00000       0.000000   
25%          0.000000        0.000000        0.00000       0.000000   
50%          0.000000        0.000000        0.00000       0.000000   
75%          0.000000        0.000000        0.00000       0.000000   
max          1.000000        1.000000        1.00000       1.000000   

        hypertension  heart_disease smoking_history            bmi  \
count   100000.00000  100000.000000          100000  100000.000000   
unique           NaN            NaN               6            NaN   
top              NaN            NaN         No Info            NaN   
freq             NaN            NaN           35816            NaN   
mean         0.07485       0.039420             NaN      27.320767   
std          0.26315       0.194593             NaN       6.636783   
min          0.00000       0.000000             NaN      10.010000   
25%          0.00000       0.000000             NaN      23.630000   
50%          0.00000       0.000000             NaN      27.320000   
75%          0.00000       0.000000             NaN      29.580000   
max          1.00000       1.000000             NaN      95.690000   

          hbA1c_level  blood_glucose_level       diabetes  
count   100000.000000        100000.000000  100000.000000  
unique            NaN                  NaN            NaN  
top               NaN                  NaN            NaN  
freq              NaN                  NaN            NaN  
mean         5.527507           138.058060       0.085000  
std          1.070672            40.708136       0.278883  
min          3.500000            80.000000       0.000000  
25%          4.800000           100.000000       0.000000  
50%          5.800000           140.000000       0.000000  
75%          6.200000           159.000000       0.000000  
max          9.000000           300.000000       1.000000

From above table we can see data is collected from year 2015 to 2022, between age 0.08 and 80, bmi between 10.01 and 95.69.

Questions:
1. What is the distribution of race?&Can a person have two different races?
2. Trend of diabetes YoY.
3. Relationship between diabetes and other factors(bmi,hypertension,heart disease,hbA1c level and bloood glucose level)?
4. Age distribution among different groups(diabetes/smoking)?
5. Location distribution of population.
6. Diabetes distribution in gender,age?

In [24]:
#Answer Q1
#First is to give each line a Patient ID
df["id"] = df.index
#Add values for all race columns
df["sum_race"] = df[["race:AfricanAmerican","race:Asian","race:Caucasian","race:Hispanic","race:Other"]].sum(axis=1)
df.describe()

year            age  race:AfricanAmerican     race:Asian  \
count  100000.000000  100000.000000         100000.000000  100000.000000   
mean     2018.360820      41.885856              0.202230       0.200150   
std         1.345239      22.516840              0.401665       0.400114   
min      2015.000000       0.080000              0.000000       0.000000   
25%      2019.000000      24.000000              0.000000       0.000000   
50%      2019.000000      43.000000              0.000000       0.000000   
75%      2019.000000      60.000000              0.000000       0.000000   
max      2022.000000      80.000000              1.000000       1.000000   

       race:Caucasian  race:Hispanic     race:Other  hypertension  \
count   100000.000000   100000.00000  100000.000000  100000.00000   
mean         0.198760        0.19888       0.199980       0.07485   
std          0.399069        0.39916       0.399987       0.26315   
min          0.000000        0.00000       0.000000       0.00000   
25%          0.000000        0.00000       0.000000       0.00000   
50%          0.000000        0.00000       0.000000       0.00000   
75%          0.000000        0.00000       0.000000       0.00000   
max          1.000000        1.00000       1.000000       1.00000   

       heart_disease            bmi    hbA1c_level  blood_glucose_level  \
count  100000.000000  100000.000000  100000.000000        100000.000000   
mean        0.039420      27.320767       5.527507           138.058060   
std         0.194593       6.636783       1.070672            40.708136   
min         0.000000      10.010000       3.500000            80.000000   
25%         0.000000      23.630000       4.800000           100.000000   
50%         0.000000      27.320000       5.800000           140.000000   
75%         0.000000      29.580000       6.200000           159.000000   
max         1.000000      95.690000       9.000000           300.000000   

            diabetes             id  sum_race  
count  100000.000000  100000.000000  100000.0  
mean        0.085000   49999.500000       1.0  
std         0.278883   28867.657797       0.0  
min         0.000000       0.000000       1.0  
25%         0.000000   24999.750000       1.0  
50%         0.000000   49999.500000       1.0  
75%         0.000000   74999.250000       1.0  
max         1.000000   99999.000000       1.0

From race above summary statistics, a person can't have more than 1 race attibute. From mean of different race, we could see the value is around 0.2. Therefore, the race distribution does not align with population race distribution. Sample is not randomly collected from the population.